src/richie/apps/search/indexers/courses.py

```diff
@@ -647,7 +647,7 @@ def get_es_document_for_course(cls, course, index=None, action="index"):
             # Pick the highlighted organization from the organizations QuerySet to benefit from
             # the prefetch of related title sets
             "organization_highlighted": {
-                title.language: title.title
+                title.language: title.menu_title if title.menu_title else title.title
```

Comment:
You need to add a test on the indexer get_es_document method or something like that